andela/wger-space
# #166185757 Reformat codebase to meet PEP8 conventions

#### What does this PR do?

Reformat the codebase to satisfy to PEP8 conventions.

#### Description of Task to be completed?

- Used the python package `black` to reformat the project files to PEPE8 standards, for instance, to limit characters on a line to 79 characters.
- Installation of the Flake8 package and added it to the requirements.txt file for testing compliance with the PEP8 standards.
- Add command to `.circleci/config.yml` to enable flake8 checks to be run by CircleCI when testing the build on a PR.

#### How should this be manually tested?

- Create a local instance of the application off this branch following the instructions in the [README](https://github.com/andela/wger-space/blob/master/README.rst) file.
- `cd` into the project root directory, the one that has the `manage.py` file.
- Run the command `flake8 wger` to use flake8 to check to style coding errors and warnings
- A successful test returns a bash prompt with no warnings, failures or errors displayed

#### Any background context you want to provide?

- The following files and folders are excluded from the flake8 style check:
 urls.py, tasks.py, migrations, bower_components, docs, settings
-  Ignored warnings, failures and errors:

- [ ] **F401** imported but unused

- [ ] **W503** line break before binary operator

- [ ] **F811** redefinition of a function

- [ ] **E741** ambiguous variable name

- The above failures, errors, and warnings were ignored because they required refactoring the code structure, which was out of the scope of this task.

#### What are the relevant pivotal tracker stories?

[#166185757](https://www.pivotaltracker.com/story/show/166185757)

#### Screenshots (if appropriate)

- A successful run of the `flake8 wger` command
<img width="1035" alt="Screen Shot 2019-05-28 at 11 19 36" src="https://user-images.githubusercontent.com/6644707/58462457-a18d7780-813a-11e9-8f3c-5e5c45ac19d6.png">
